### PR TITLE
Cmake improvements

### PR DESCRIPTION
**Pull request by scribam**
Some improvements/cleanup for the CMake build system + cotire 1.8.0

**Changelog for cotire 1.8.0**
- support for clang-cl.exe under Windows.
- faster prefix header generation for Clang.
- enable parallel compilation of unity target for MSVC.
- CMake 3.9 and 3.10 compatibility fixes.
- disable inclusion of timestamp in precompiled headers for Clang.
- work around ccache reporting incorrect configuration.
- honor MANUALLY_ADDED_DEPENDENCIES property upon generation of unity targets.
- use default setting of 2 for COTIRE_MINIMUM_NUMBER_OF_TARGET_SOURCES.
- drop broken support for unity building of targets with automatic CMake Qt processing turned on.
- manual updates.

### CODE REVIEW

File: rpcs3/CMakeLists.txt [hunk 1]
```diff
@@ -124,7 +128,7 @@ if(NOT MSVC)
 
 	add_compile_options(-msse -msse2 -mcx16 -mrtm)
 
-	if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+	if(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
```
Comment (Orphis) [inline rpcs3/CMakeLists.txt]: This is a case of double expansion, drop the ${} around the variable name.

File: rpcs3/CMakeLists.txt [hunk 2]
```diff
@@ -107,7 +107,11 @@ if(USE_NATIVE_INSTRUCTIONS)
 endif()
 
 if(NOT MSVC)
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-invalid-pch -Wno-unused-command-line-argument -fexceptions")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-invalid-pch -fexceptions")
+	if(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
```
Comment (Orphis) [inline rpcs3/CMakeLists.txt]: This is a case of double expansion, drop the ${} around the variable name.

File: rpcs3/CMakeLists.txt (hunk 1 above)
Comment (scribam) [reply rpcs3/CMakeLists.txt]: Indeed, thank you for the review

File: rpcs3/CMakeLists.txt (hunk 2 above)
Comment (scribam) [reply rpcs3/CMakeLists.txt]: Fixed

**Pull request merged**